components/language-chooser/common/find-language/matchingSubstringDemarcation.ts

```diff
@@ -60,7 +60,7 @@ export function deepStripDemarcation<T>(demarcated: T): T {
     return demarcated.map((element) => deepStripDemarcation(element)) as T;
   }
   if (typeof demarcated === "object") {
-    const newObject = {} as any;
+    const newObject: any = {};
```

Comment:
Are these anys okay? Also on line 65. Would it be better to create a new interface e.g. 
`
interface Foo<T> {
  [key: string]: T;
} and if so what should it be called? Seems like a lot of code for something so small?

Reply:
Ok with me.
<!-- Reviewable comment -OBQnLU36VFOEfp8z9wr:b6apnsr -->
<!-- Sent from Reviewable.io -->